test/axelar-gateway.js

```diff
@@ -0,0 +1,184 @@
+    let operatorKeys;
+    let signers;
+    let nonce = 0;
+    let packageId;
+
+    function calculateNextSigners() {
+        operatorKeys = [getRandomBytes32(), getRandomBytes32(), getRandomBytes32()];
+        pubkeys = operatorKeys.map((key) => Secp256k1Keypair.fromSecretKey(arrayify(key)).getPublicKey().toRawBytes());
+        const keys = operatorKeys.map((key, index) => {
+            return { privkey: key, pubkey: pubkeys[index] };
+        });
+        keys.sort((key1, key2) => {
+            for (let i = 0; i < 33; i++) {
+                if (key1.pubkey[i] < key2.pubkey[i]) return -1;
+                if (key1.pubkey[i] > key2.pubkey[i]) return 1;
+            }
+
+            return 0;
+        });
+        operatorKeys = keys.map((key) => key.privkey);
+        signers = {
+            signers: keys.map((key) => {
+                return { pubkey: key.pubkey, weight: 1 };
+            }),
+            threshold: 2,
+            nonce: hexlify([++nonce]),
+        };
+    }
+
+    function hashMessage(data) {
+        const toHash = new Uint8Array(data.length + 1);
+        toHash[0] = COMMAND_TYPE_ROTATE_SIGNERS;
+        toHash.set(data, 1);
+
+        return keccak256(toHash);
+    }
+
+    function sign(privKeys, messageToSign) {
+        const signatures = [];
+
+        for (const privKey of privKeys) {
+            const { signature, recid } = secp256k1.ecdsaSign(arrayify(keccak256(messageToSign)), arrayify(privKey));
+            signatures.push(new Uint8Array([...signature, recid]));
+        }
+
+        return signatures;
+    }
+
+    async function sleep(ms = 1000) {
+        await new Promise((resolve) => setTimeout(resolve, ms));
+    }
+
+    const minimumRotationDelay = 1000;
+
+    before(async () => {
+        client = new SuiClient({ url: getFullnodeUrl('localnet') });
+
+        await Promise.all(
+            [operator, deployer, keypair].map((keypair) =>
+                requestSuiFromFaucetV0({
+                    host: getFaucetHost('localnet'),
+                    recipient: keypair.toSuiAddress(),
+                }),
+            ),
+        );
+
+        let result = await publishPackage(client, deployer, 'axelar_gateway');
+        packageId = result.packageId;
+        const creatorCap = result.publishTxn.objectChanges.find(
+            (change) => change.objectType === `${packageId}::gateway::CreatorCap`,
+        ).objectId;
+
+        calculateNextSigners();
+
+        const encodedSigners = axelarStructs.WeightedSigners.serialize(signers).toBytes();
+        const builder = new TxBuilder(client);
+
+        const separator = await builder.moveCall({
+            target: `${packageId}::bytes32::new`,
+            arguments: [domainSeparator],
+        });
+
+        await builder.moveCall({
+            target: `${packageId}::gateway::setup`,
+            arguments: [creatorCap, operator.toSuiAddress(), separator, minimumRotationDelay, encodedSigners, '0x6'],
```

Comment:
avoid magic constants
```suggestion
            arguments: [creatorCap, operator.toSuiAddress(), separator, minimumRotationDelay, encodedSigners, suiClock],
```